pages/accountLists/[accountListId]/tools/csv.page.test.tsx

```diff
@@ -0,0 +1,220 @@
+          <TestRouter router={router}>
+            <CsvHome></CsvHome>
+          </TestRouter>
+        </TestWrapper>,
+      );
+
+      expect(await findByTestId('MaxFileSize')).toBeVisible();
+    });
+
+    it('should show the headers tab if specified in the URL', async () => {
+      const router = buildRouter(CsvImportViewStepEnum.Headers);
+      renderCsvHome(router);
+
+      await waitFor(() => {
+        expect(router.replace).toHaveBeenCalledWith(
+          expect.objectContaining({
+            query: {
+              accountListId,
+              tab: CsvImportViewStepEnum.Headers,
+              ...{ id: csvFileId },
+            },
+          }),
+        );
+      });
+    });
+
+    it('should show the headers tab if specified', async () => {
+      const { findByRole } = renderCsvImportWrapper(
+        CsvImportViewStepEnum.Headers,
+      );
+
+      expect(
+        await findByRole('columnheader', { name: 'Your CSV Header' }),
+      ).toBeVisible();
+    });
+
+    it('should show the values tab if specified in the URL', async () => {
+      const router = buildRouter(CsvImportViewStepEnum.Values);
+      renderCsvHome(router);
+
+      await waitFor(() => {
+        expect(router.replace).toHaveBeenCalledWith(
+          expect.objectContaining({
+            query: {
+              accountListId,
+              tab: CsvImportViewStepEnum.Values,
+              ...{ id: csvFileId },
+            },
+          }),
+        );
+      });
+    });
+
+    it('should show the values tab if specified', async () => {
+      uploadData.valuesToConstantsMappings = {
+        test: 'wee',
+      };
+      initialData = { ...uploadData };
+      const { findByRole } = renderCsvImportWrapper(
+        CsvImportViewStepEnum.Values,
+      );
+
+      expect(
+        await findByRole('columnheader', { name: 'Your CSV value' }),
+      ).toBeVisible();
+    });
+
+    it('should show the preview tab if specified in the URL', async () => {
+      const router = buildRouter(CsvImportViewStepEnum.Preview);
+      renderCsvHome(router);
+
+      await waitFor(() => {
+        expect(router.replace).toHaveBeenCalledWith(
+          expect.objectContaining({
+            query: {
+              accountListId,
+              tab: CsvImportViewStepEnum.Preview,
+              ...{ id: csvFileId },
+            },
+          }),
+        );
+      });
+    });
+
+    it('should show the preview tab if specified', async () => {
+      uploadData.sampleContacts = [{} as CsvContact];
+      const { findByRole } = renderCsvImportWrapper(
+        CsvImportViewStepEnum.Preview,
+      );
+
+      expect(
+        await findByRole('columnheader', { name: 'Contact Name' }),
+      ).toBeVisible();
+    });
+
+    const buildRouter = (tab) => {
+      return {
+        isReady: true,
+        query: {
+          tab: tab,
+          id: csvFileId,
+        },
+        replace: jest.fn(),
+      };
+    };
+
+    const renderCsvHome = (router) => {
```

Comment:
Yeah, a preference thing. This actually isn't defining a component, but a helper function used by a subset of tests. In the Java world we put such things below the last usage of the function (method). However, I don't mind putting it at the top if that is the consistent way to do it across MPDx.

Reply:
The code style in Java is mentioned [here](https://docs.google.com/document/d/16TGhx6nL3d3cjN8Rmpo-oY55ErTSUT6sZx8RVDTpgJU/edit#heading=h.pvhozz639k43). I don't think we have an equivalent for JS.

Reply:
In MPDX, we normally define helper functions before they are used as function expressions will cause an error if you try to define them after they're used. However, you can define function declarations, as JS can hoist them. This might be similar to Java. idk.


**Function Declarations**
Function declarations are hoisted, which means that you can use a function before it is defined in your code. For example:
```
console.log(add(2, 3)); // 5

function add(a, b) {
  return a + b;
}
```
In this case, the add function is declared later in the code, but it can still be called earlier because the function declaration is hoisted to the top of the scope.

**Function Expressions**
Function expressions, including arrow functions and anonymous functions assigned to variables, are not hoisted. Therefore, you must define them before you use them. For example:
```
// console.log(multiply(2, 3)); // Error: multiply is not a function

const multiply = function (a, b) {
  return a * b;
};

console.log(multiply(2, 3)); // 6
```
In this case, attempting to call multiply before it is defined would result in an error.


While function declarations are hoisted and can be defined after they are used, it's generally a good practice to define functions before they are used.

Reply:
Are we normally defining them just before they are used the first time or at the top of the file?